src/lazy/value_ref.rs

```diff
@@ -23,7 +24,7 @@ pub enum ValueRef<'top, 'data, D: LazyDecoder<'data>> {
     Timestamp(Timestamp),
     String(StrRef<'data>),
     Symbol(SymbolRef<'top>),
-    Blob(&'data [u8]),
+    Blob(BytesRef<'data>),
```

Comment:
🗺️ Like `RawValueRef`, `ValueRef` now returns a `BytesRef` instead of a `&[u8]` so it has the option to allocate a `Vec<u8>` when the input encoding is base64 text. The binary reader can still return a slice of the input buffer.